core/src/main/java/com/linecorp/armeria/server/Route.java

```diff
@@ -0,0 +1,127 @@
+        return new RouteBuilder();
+    }
+
+    /**
+     * Matches the specified {@link RoutingContext} and extracts the path parameters from it if exists.
+     *
+     * @param routingCtx a context to find the {@link Service}
+     *
+     * @return a non-empty {@link RoutingResult} if the {@linkplain RoutingContext#path() path},
+     *         {@linkplain RoutingContext#method() method},
+     *         {@linkplain RoutingContext#contentType() contentType} and
+     *         {@linkplain RoutingContext#acceptTypes() acceptTypes} matches the equivalent conditions in
+     *         {@link Route}. {@link RoutingResult#empty()} otherwise.
+     *
+     * @see RouteBuilder#methods(Iterable)
+     * @see RouteBuilder#consumes(Iterable)
+     * @see RouteBuilder#produces(Iterable)
+     */
+    RoutingResult apply(RoutingContext routingCtx);
+
+    /**
+     * Returns the names of the path parameters extracted by this mapping.
+     */
+    Set<String> paramNames();
+
+    /**
+     * Returns the logger name.
+     *
+     * @return the logger name whose components are separated by a dot (.)
+     */
+    String loggerName();
+
+    /**
+     * Returns the value of the {@link Tag} in a {@link Meter} of this {@link Route}.
+     */
+    String meterTag();
+
+    /**
+     * Returns the exact path of this {@link Route} if this has an exact path matching condition,
+     * {@link Optional#empty()} otherwise.
+     */
+    Optional<String> exactPath();
+
+    /**
+     * Returns the prefix of this {@link Route} if this has a prefix matching condition,
+     * {@link Optional#empty()} otherwise.
+     *
+     * @return the prefix that ends with '/' if this {@link Route} has a prefix matching condition
+     */
+    Optional<String> prefix();
+
+    /**
+     * Returns the trie path of this {@link Route} if this has a trie matching condition,
+     * {@link Optional#empty()} otherwise. The trie path does not exist when the {@link Route} is created
+     * using a glob or a regex expression.
+     *
+     * @see RouteBuilder#glob(String)
+     * @see RouteBuilder#regex(String)
+     */
+    Optional<String> triePath();
+
+    /**
+     * Returns the regular expression of this {@link Route} if this is created with the regular expression or
+     * glob pattern. Please note that this regex does not include the {@code pathPrefix} if this is
+     * created with {@link RouteBuilder#pathWithPrefix(String, String)}.
+     * You should call {@link #prefix()} to retrieve that.
+     */
+    Optional<String> regex();
+
+    /**
+     * Returns the complexity of this {@link Route}. It would be increased if this path mapping has
+     * more conditions to check.
+     */
+    int complexity();
+
+    /**
+     * Returns the {@link Set} of {@link HttpMethod}s that this {@link Route} supports.
+     */
+    Set<HttpMethod> methods();
+
+    /**
+     * Returns the {@link List} of {@link MediaType}s that this {@link Route} consumes.
```

Comment:
I guess it might just be for symmetry, but just wondering does this need to be a `List` instead of `Set`? I think requests only have one content type so this also can only have one possible match per request (unlike produces)

Reply:
The `HttpHeaderPathMapping` which is sort of the ancestor of this class, had the types in a `List`. But yeah, it should be a `Set` because this indicates the content types that the mapped service accepts.
And the produces should be a `Set` as well. Let me make changes. Thanks!